insights/parsers/kpatch_patches.py

```diff
@@ -0,0 +1,40 @@
+"""
+KpatchPatches - report locally stored kpatch patches
+====================================================
+
+This parser creates a list of the module names of locally
+stored kpatch modules. If no modules are installed, the
+list will be empty.
+
+"""
+
+from .. import parser, Parser
+from insights.specs import Specs
+import re
+
+
+@parser(Specs.kpatch_patch_files)
+class KpatchPatches(Parser):
+    """
+    A parser for getting locally stored kpatch-patch modules
+    """
+    __re_cannot_access = re.compile(r"No such file or directory$")
+    __re_archive = re.compile(r"\.xz$")
+
+    def get_patches(self):
+        return self.patches
+
+    def parse_content(self, content):
+        self.patches = content
+        # If kpatch is not installed, the directory will not exist.
+        # In that case, the list of local kpatch files is empty.
+        if self.__re_cannot_access.search(content[0]):
+            self.patches = []
```

Comment:
Better `raise SkipException` here for no kpatch is installed, rasing the SkipException will avoid generating an instance for the Parser. Then you won't need to do further check when using it (`if obj.patches: ...`.

Reply:
I wasn't quite sure where the appropriate place to raise the SkipException was, so I added an __init__ function for the parser that does a try/catch on super()__init__ and raises a SkipException when it catches a ContentException. Is this the right way to do it?

In any case, I remove this check from parse_content() in the latest patch.